openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java

```diff
@@ -598,12 +598,14 @@ private List<GenSrcFile> generateBallerinaService(Path openAPI, String serviceNa
         String concatTitle = serviceName.toLowerCase(Locale.ENGLISH);
         String srcFile = concatTitle + "_service.bal";
         OpenAPI openAPIDef = normalizeOpenAPI(openAPI, false);
-        BallerinaServiceGenerator ballerinaServiceGenerator = new BallerinaServiceGenerator(openAPIDef, filter);
+        BallerinaServiceGenerator ballerinaServiceGenerator = BallerinaServiceGenerator.getInstance();
+        ballerinaServiceGenerator.initialize(openAPIDef, filter);
```

Comment:
Yeah, there is a requirement to pass some object instance data to schema generators that are generated using the service. Therefore, to avoid some thread conflicting issues used the singleton pattern since the service generator is only a single instance of a class that is required to execute. We need some initialized data to proceed with code and since it has a private constructor here used public initialize method to initialize the instance with the required data.